lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -348,7 +355,17 @@ def _on_start(self, event: StartEvent):
 
         # request the start of OpenSearch
         self.status.set(WaitingStatus(RequestUnitServiceOps.format("start")))
-        self._start_opensearch_event.emit()
+
+        deployment_desc = self.opensearch_peer_cm.deployment_desc()
+        ignore_lock = (
+            "data" not in deployment_desc.config.roles
+            and deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR
+            or "data" in deployment_desc.config.roles
+            and self.unit.is_leader()
+            and deployment_desc.typ == DeploymentType.OTHER
```

Comment:
This is only happening if the security index is not initialized and the unit is the leader, so is only valid for the first data node to start.

Reply:
@reneradoi @Mehdi-Bendriss what happens if we have:
* MAIN_ORCHESTRATOR: CM only
* FAILOVER_ORCHESTRATOR: CM + single data cluster

I think, in this case, we'd never initialize the data cluster?

Reply:
```suggestion
            and deployment_desc.typ in [DeploymentType.OTHER, DeploymentType.FAILOVER]
```

Reply:
> @reneradoi @Mehdi-Bendriss what happens if we have:
> 
> * MAIN_ORCHESTRATOR: CM only
> * FAILOVER_ORCHESTRATOR: CM + single data cluster
> 
> I think, in this case, we'd never initialize the data cluster?

Thanks for this comment @phvalguima it's a really good question! Thinking about the code again, I came to the conclusion that the start and initialization of this failover-node (cluster manager + data roles) will not be prevented. The added logic does not impact the existing capabilities to start such nodes, so the main-orchestrator that was started first should wait with starting for this node, but once the failover has been added, they should just form the cluster and that's it.

The logic here is only to allow non-cm-nodes to start without acquiring the lock, this will not prevent starting "regular" cm+data nodes.

We should add test coverage for that, but I can only do this next week because I will travel back now.

Reply:
Could we just have a test for this case.... _Please_...?

Reply:
@reneradoi ack, I will attach this discussion to our sync-up ticket.